testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```
@@ -724,119 +724,85 @@ async fn test_delayed_decryption_latest_event() -> Result<()> {
+    let bob_sync_service = SyncService::builder(bob.clone()).build().await.unwrap();
+    bob_sync_service.start().await;
 
-    // alice creates a room and invites bob.
-    let alice_room = alice
+    // Alice creates a room and invites Bob.
+    let room = alice
         .create_room(assign!(CreateRoomRequest::new(), {
             invite: vec![bob.user_id().unwrap().to_owned()],
             is_direct: true,
             preset: Some(RoomPreset::TrustedPrivateChat),
         }))
         .await?;
-    alice_room.enable_encryption().await.unwrap();
 
-    sleep(Duration::from_secs(1)).await;
-    let alice_room = alice.get_room(alice_room.room_id()).unwrap();
+    // Room is created by Alice. Let's enable encryption.
+    room.enable_encryption().await.unwrap();
+
+    // Wait for Alice to see its new room, and for Bob to receive the invitation.
+    sleep(Duration::from_secs(2)).await;
+
+    let alice_room = alice.get_room(room.room_id()).unwrap();
     let bob_room = bob.get_room(alice_room.room_id()).unwrap();
     bob_room.join().await.unwrap();
 
-    sleep(Duration::from_secs(1)).await;
+    // Wait for Alice to see Bob in the room.
+    sleep(Duration::from_secs(2)).await;
+
     assert_eq!(alice_room.state(), RoomState::Joined);
     assert!(alice_room.is_encrypted().await.unwrap());
+
     assert_eq!(bob_room.state(), RoomState::Joined);
+    assert!(bob_room.is_encrypted().await.unwrap());
 
+    // Get the room list of Alice.
     let alice_all_rooms = alice_sync_service.room_list_service().all_rooms().await.unwrap();
-    let (stream, entries) = alice_all_rooms
+    let (alice_room_list_stream, entries) = alice_all_rooms
         .entries_with_dynamic_adapters(10, alice.room_info_notable_update_receiver());
     entries.set_filter(Box::new(new_filter_all(vec![])));
-    pin_mut!(stream);
+    pin_mut!(alice_room_list_stream);
+
+    // The room list receives its initial rooms.
+    assert_let!(Some(diffs) = alice_room_list_stream.next().await);
```

general nit: prefer adding a timeout do those waits, to make the tests fail fast in real-world conditions